src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/RazorCodeGenerationOptions.cs

```diff
@@ -113,56 +140,104 @@ public bool SuppressAddComponentParameter
     public bool RemapLinePragmaPathsOnWindows
         => _flags.HasFlag(RazorCodeGenerationOptionsFlags.RemapLinePragmaPathsOnWindows);
 
-    public static RazorCodeGenerationOptions Default { get; } = new RazorCodeGenerationOptions(
-        flags: RazorCodeGenerationOptionsFlags.DefaultFlags,
-        indentSize: 4,
-        newLine: Environment.NewLine,
-        rootNamespace: null,
-        suppressUniqueIds: null);
+    public RazorCodeGenerationOptions WithIndentSize(int value)
+        => IndentSize == value
+            ? this
+            : new(value, NewLine, RootNamespace, SuppressUniqueIds, _flags);
+
+    public RazorCodeGenerationOptions WithNewLine(string value)
+        => NewLine == value
+            ? this
+            : new(IndentSize, value, RootNamespace, SuppressUniqueIds, _flags);
+
+    public RazorCodeGenerationOptions WithRootNamespace(string? value)
+        => RootNamespace == value
+            ? this
+            : new(IndentSize, NewLine, value, SuppressUniqueIds, _flags);
+
+    public RazorCodeGenerationOptions WithSuppressUniqueIds(string? value)
+        => RootNamespace == value
+            ? this
+            : new(IndentSize, NewLine, RootNamespace, value, _flags);
+
+    public RazorCodeGenerationOptions WithFlags(
+        Optional<bool> designTime = default,
```

Comment:
What is the advantage of using `MS.CA.Optional<T>` here over `Nullable<T>`/`bool?`

Reply:
Functionally, there isn't any added value for this particular method. The reason I reached for it is because `Optional<T>` was designed for this purpose and it expresses the intent better. `Optional<T>` intended to represent a value that might be present or not and `null` could be a valid value. `Nullable<T>` was designed specifcally for struct values that could be `null` or not (a la old school `DBNull`).